File: console/src/types/risk.ts
export interface RiskKpi {
  label: string
  value: string
  hint: string
  kind: string | null
}

export interface CapitalBucket {
  name: string
  used: number
  cap: number
  state: string
  kind: string
}

export interface RiskCap {
  label: string
  value: string
  hint: string
}

export interface KillSwitch {
  name: string
  state: string
  trigger: string
  kind: string
}

export interface RiskResponse {
  kpis: RiskKpi[]
  buckets: CapitalBucket[]
  caps: RiskCap[]
  kill_switches: KillSwitch[]
}

File: api/models/risk.py
from __future__ import annotations

from pydantic import BaseModel


class RiskKpi(BaseModel):
    label: str
    value: str
    hint: str
    kind: str | None = None


class CapitalBucket(BaseModel):
    name: str
    used: float
    cap: float
    state: str
    kind: str


class RiskCap(BaseModel):
    label: str
    value: str
    hint: str


class KillSwitch(BaseModel):
    name: str
    state: str
    trigger: str
    kind: str


class RiskResponse(BaseModel):
    kpis: list[RiskKpi]
    buckets: list[CapitalBucket]
    caps: list[RiskCap]
    kill_switches: list[KillSwitch]
